base/range.jl

```diff
@@ -253,6 +253,8 @@ end
 # For types where x+oneunit(x) may not be well-defined
 steprange_last_empty(start, step, stop) = start - step
 
+StepRange{T,S}(start, step, stop) where {T,S} = StepRange{T,S}(T(start), S(step), T(stop))
```

Comment:
Nice. But I think there's an even better solution: take the type annotations off the arguments in https://github.com/JuliaLang/julia/blob/5f0469b2273c8de57a26aec88468e5b5eed3a61b/base/range.jl#L204

Anyone who calls the constructor with specific type-parameters is guaranteeing what the result type should be, and there aren't multiple inner constructors, so I think there's no need to restrict the types of the arguments.

Reply:
Done! The values are still converted explicitly for `steprange_last`, so that one can do `StepRange{Int,Int}(1.0, 2.0, 3.0)` without getting an error.

Reply:
Test failures seem related. Perhaps my suggestion led you in the wrong direction?

Reply:
Should be fixed now.